notebooks/ch12/linear_projection.md

```diff
@@ -0,0 +1,567 @@
+&= 
+\left(
+\begin{bmatrix} 3 & 2\\ 2 & 10 \end{bmatrix}
+\right)^{-1}
+\begin{bmatrix} 1 \\ 8 \end{bmatrix} \\
+&=
+\frac{1}{30-4}
+\begin{bmatrix} 10 & -2\\ -2 & 3 \end{bmatrix}
+\begin{bmatrix} 1 \\ 8 \end{bmatrix} \\
+&=
+\frac{1}{26}
+\begin{bmatrix} -6 \\ 22 \end{bmatrix}\\
+&=
+\begin{bmatrix} - \frac{3}{13} \\ \frac{11}{13} \end{bmatrix}
+\end{align}
+$$
+
+We have analytically found that best model for least squares regression is $f_\hat{\theta} (x_i) = - \frac{3}{13} + \frac{11}{13} x_i$. We know that our choice of $\hat{\theta}$ is sound by the mathematical property that the projection of $\vec{y}$ onto the span of the columns of $X$ yields the closest point in the vector space to $\vec{y}$. Under linear constraints using the least squares metric, solving for $\hat{\theta}$ by taking the projection guarantees us the optimal solution.
+
+## When Variables are Linearly Dependent
+
+For every additional variable, we add one column to $X$. The span of the columns of $X$ is the linear combinations of the column vectors, so adding variables only changes the span into a higher dimensional subspaces when the added variable is linearly independent of the existing ones. Otherwise, if a column is linearly dependent, it can be expressed as a linear combination of some other column. Thus, for every linear combination it contributes to, we can replace it with a linear combination of the other columns. So we have not expanded the set of possible vectors spanned by the original columns of $X$.
+
+Recall that the span of $X$ is important because it is the subspace we want to project $\vec{y}$ onto. If the subspace does not change, then the projection will not change.
+
+For example, when we introduced $\vec{x}$ when moving from the constant model to linear model, we introduced a independent variable: $\vec{x} = \begin{bmatrix} 3 \\ 0 \\ -1 \end{bmatrix}$ cannot be expressed as a scalar of $\begin{bmatrix} 1 \\ 1 \\ 1 \end{bmatrix}$.
+
+However, say we instead introduced a new variable $d$ with values:
+
+| d | y |
+| - | :-|
+| -0.5 | 2 |
+| -0.5 | 1 |
+| -0.5 | -2 |
+
+Then, we would be minimizing the loss function:
+
+$$ 
+\begin{aligned}
+L(\hat{\theta}, \vec{d}, \vec{y})
+&= \left \Vert  \qquad   
+\begin{bmatrix} y_1 \\ y_2 \\ y_3  \end{bmatrix} \quad - \quad 
+\begin{bmatrix} 1 & d_1 \\ 1 & d_2 \\ 1 & d_3 \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix}
+\qquad \right \Vert ^2 \\
+&= \left \Vert  \qquad   
+\begin{bmatrix} y_1 \\ y_2 \\ y_3  \end{bmatrix} \quad - \quad 
+\begin{bmatrix} 1 & -0.5 \\ 1 & -0.5 \\ 1 & -0.5 \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix}
+\qquad \right \Vert ^2 \\
+&= \left \Vert  \qquad  
+\vec{y} 
+\quad - \quad 
+X 
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix}
+\qquad \right \Vert ^2 \\
+&= \left \Vert  \qquad  
+\vec{y} 
+\quad - \quad 
+\hat{y}
+\qquad \right \Vert ^2 \\
+\end{aligned}
```

Comment:
Hmm.. not sure we have to re-do the derivation here :O

Reply:
taken out 👍